extra/3RD_PARTY_QUICK_START.md

```diff
@@ -7,24 +7,177 @@ So you'd like to create and share your own language for Highlight.js.  That's aw
+
 
 - Checkout highlight-js from github...
-- 3rd party languages are placed into the `extra` directory
 
-So if you had a `xyz` language you'd create an `extra/xyz` folder, and that would be your language module. All paths below are relative to that.
+- Create directory `extra/highlightjs-xyz`. This is home directory of your extension. Init repository there, and `package.json`. Here is the structure of your folder.
+
+    ```
+    - dist
+    - - xyz.min.js
+    - src
+    - - xyz.js
+    - test
+    - - detect
+    - - - detect.txt
+    - - markup
+    - - - sample.txt
+    - - - sample.expect.txt
+    - - index.js
+    - package.json
+    - LICENSE
+    - README.md
+    ```
+
+    And other project files like `.gitignore`, `CHANGES.md` and so on..
+
+    
+- Init NPM package
+
+    Create `package.json` file. You can do that by running command inside you `highlightjs-xyz` directory.
+
+    ```
+    npm init
+    ```
+
+    Then add there dependencies and scripts. Here is how your file might look like.
+
+    ```json
+    {
+        "name": "highlightjs-xyz",
+        "version": "0.0.1",
+        "description": "XYZ",
+        "main": "src/xyz.js",
+        "scripts": {
+            "postpublish": "PACKAGE_VERSION=$(cat package.json | grep \\\"version\\\" | head -1 | awk -F: '{ print $2 }' | sed 's/[\",]//g' | tr -d '[[:space:]]') && git tag v$PACKAGE_VERSION && git push --tags",
+            "test": "./node_modules/.bin/mocha --reporter spec"
+        },
+        "files": [
+            "src/xyz.js",
+            "dist/xyz.min.js"
+        ],
+        "devDependencies": {
+            "highlight.js": "^10.4.1",
+            "mocha": "^8.3.2",
+            "should": "^13.2.3",
+        }
+    }
+    ```
+
+    Now run command
+
+    ```
+    npm install
+    ```
+- Create your source file
+
+    Create file `src/xyz.js`. Something like this.
+
+    ```js
+    /*
+    * highlight.js XYZ syntax highlighting definition
+    *
+    * @see https://github.com/highlightjs/highlight.js
+    *
+    * @package: highlightjs-xyz
+    * @author:  Jhon Smith <johm@smith.com>
+    * @since:   2019-08-02
+    *
+    * Description: XYZ is new language.
+    * Category: scripting
+    */
+
+    module.exports = function (hljs) {
+        return {
+            aliases: ["xyz"],
+            name: "XYZ",
+            case_insensitive: true,
+            keywords: {
+                keyword: "if else",
+                title: "program function",
+                literal: "false true null ",
+                built_in: "array pointer",
+                function: "mod abs acos asin atan",
+            },
+            contains: [
+                hljs.C_NUMBER_MODE,
+                hljs.COMMENT("//", "$"),
+                hljs.C_BLOCK_COMMENT_MODE,
+                hljs.COMMENT("\\/\\*", "\\*\\/"),
+            ],
+        };
+    };
+    ```
+
+    This is just a bootstrap, of course follow documentation to add extensive syntax highlight rules.
+
+
+## Test
```

Comment:
Please remove this section for now (you can save it locally for perhaps a future PR).  We're still discussing this and there is overall here with the new template repository... We are not (yet) deprecating how testing is currently done.